aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -3,66 +3,76 @@ module AptosFramework::Stake {
+    const EALREADY_REGISTERED: u64 = 10;
+    /// Account does not have the right ownership capability.
+    const ENOT_OWNER: u64 = 11;
+    /// No coins in inactive state to withdraw from specified pool.
+    const ENO_COINS_TO_WITHDRAW: u64 = 12;
+    /// Account does not have the right operator capability.
+    const ENOT_OPERATOR: u64 = 13;
+
+    /// Capability that represents ownership and can be used to control the validator and the associated stake pool.
+    /// Having this be separate from the signer for the account that the validator resources are hosted at allows
+    /// modules to have control over a validator.
+    struct OwnerCapability has key, store {
+        pool_address: address,
     }
 
-    /// Aggregation of delegation and represent a validator's voting power, stored in ValidatorInfo.
-    /// Invariants:
-    /// 1. voting_power = sum(active + pending_inactive)
-    /// 2. user interact with pending_active and inactive if it's in the ValidatorSet.
-    /// 3. user interact with active, inactive if it's not in the ValidatorSet.
-    /// 4. pending_active and pending_inactive are empty if it's not in the ValidatorSet.
-    struct StakePool has key, store {
-        // sum of active and pending_inactive stakes, updated on epoch boundary for active validators.
-        voting_power: u64,
-        // sum of active and pending_active stakes, updated when delegate/unlock happens.
-        next_epoch_voting_power: u64,
+    /// Each validator has a separate StakePool resource.
+    ///
+    /// Similar to validator set, any changes to stake for an active validator only take effect in the next epoch.
+    ///
+    /// The validator's stake can be in different state, as
+    /// 1. Current epoch's voting_power = active + pending_inactive
+    /// 2. Next epoch's voting_power = active + pending_active
+    /// 3. If the validator is already active, changes in stake are applied on pending_active or pending_inactive.
+    /// 4. Otherwise, changes in stake can be written directly to active/inactive.
+    /// 5. Only inactive stake can be withdrawn.
+    struct StakePool has key {
         // active stake
-        active: IterableTable<address, Delegation>,
+        active: Coin<TestCoin>,
         // inactive stake, can be withdrawn
-        inactive: IterableTable<address,Delegation>,
+        inactive: Coin<TestCoin>,
         // pending activation for next epoch
-        pending_active: IterableTable<address, Delegation>,
+        pending_active: Coin<TestCoin>,
         // pending deactivation for next epoch
-        pending_inactive: IterableTable<address, Delegation>,
+        pending_inactive: Coin<TestCoin>,
+        locked_until_secs: u64,
+        // Track the current operator of the validator node.
+        // This allows the operator to be different from the original account and allow for separation of
+        // the validator operations and ownership.
+        operator_address: address,
```

Comment:
I'm not convinced that this is the right behavior. The account handling as of today is already separated... a validator need not be managed by a hot wallet account. I'd caution against this ... just yet.

Reply:
This is actually required to separate staking from delegation. The idea is the fund owner (can be another account or staking service) can deposit coins with a node operator who is only responsible for operating the node (join validation set/validate/leave validator set) rather than having control of the funds.

Reply:
Per offline discussion, https://github.com/aptos-labs/aptos-core/pull/1082 is the built-in delegations prototype and is a lot more complex than this staking-only model. We'll go with this staking-only model for simplicity and flexibility, as staking services can easily be built on it.